### ISSUE TITLE
Allow regtest to be used for development

### ISSUE THREAD

**Issue by igorsyl** [blockstack/blockstack-app#3492]
sBTC WG is targeting a Hackathon around June 1st and we would like to allow hackers to easily build sBTC Apps using Clarinet and Hiro Wallet.

cc @andrerserrano @kenrogers

**Comment by markmhendrickson** [comment]
@kyranjamie any idea if this is a big or small lift, and whether an outside PR would be helpful?

**Comment by igorsyl** [comment]
If outside PR is an option, @andrerserrano proposed making a hackathon complete this.

**Comment by kenrogers** [comment]
@kyranjamie correct me if I'm wrong, but I believe we just need to add the correct prefix to this file: https://github.com/hirosystems/wallet/blob/dev/src/shared/crypto/bitcoin/bitcoin.network.ts along with updating some of the relevant files that rely on this configuration, the UI is one thing I haven't looked at though so not sure what the lift there would be. 

Bech32 addresses are different on testnet and regtest, which is annoying. Bitcoin-JS added support for them here if that helps as a reference: https://github.com/bitcoinjs/bitcoinjs-lib/blob/5137976e7b420500aaf37af52919d301b9b94c9b/src/networks.js#L15

And here's a long thread on the issue: https://github.com/bitcoin/bitcoin/issues/12314

Happy to contribute to this PR as well if you want to point me in the right direction, but also may be faster for someone from the Hiro team to just knock it out unless I'm mistaken on the lift, just let me know how I can be most helpful.

**Comment by kyranjamie** [comment]
Thanks for looking into it @kenrogers. We've also wanted to add signet for other use cases, would suggest we add both while we're at it,.

Two main parts to this:
1) Adding UI to allow switching between testnet/regtest/signet cc/ @mica000 
2) Modelling the new network interfaces. There's two ways to do this, not sure which is best
  a. Expanding `NetworkModes` for bitcoin to be `'mainnet' | 'testnet' | 'regtest' | 'signet'`, or
  b. Creating a separate `type TestnetModes = 'testnet' | 'regtest' | 'signet'`

Erring on the side of the latter. Open to outside contributions for this, but if above changes causes too many type issues, might be better for someone in house to finish.

**Comment by kenrogers** [comment]
Okay @kyranjamie I'll create a PR for this to get part 2 going and get as far as I can. Happy to help contribute to the UI as well but not sure if you need to go through a design phase first

**Comment by markmhendrickson** [comment]
Presumably design-wise we just need to add another option here for "Regtest"?

<img width="396" alt="Screenshot 2023-03-29 at 13 07 16" src="https://user-images.githubusercontent.com/28991/228515068-9d4d8091-e069-4539-b42c-b1c7a67857b6.png">

**Comment by kenrogers** [comment]
Can we even just have this new regtest option be the default when we switch to devnet, without needing to add any ui?

Then we can add ui for choosing between testnet and signet when someone selects testnet from the above menu.

Selfishly, I'd like to get regtest added asap for a sBTC tutorial I'm working on and I don't think we actually need any ui to make that the default option when selecting devnet, then as a separate PR we can add ui support for signet and switching Bitcoin networks.

Let me know your thoughts on that option @markmhx and @kyranjamie

**Comment by kyranjamie** [comment]
@markmhx's screenshot: I would better imagine it to be more like a radio button under testnet networks, rather than a new item in this list entirely.

> Can we even just have this new regtest option be the default when we switch to devnet, without needing to add any ui?

This could be a first step. Welcome those changes.

**Comment by kenrogers** [comment]
@kyranjamie I've made what I think are most of the changes needed to get DevNet working. It looks like a change is also required to the @stacks/transactions package to add the ChainID.  The wallet seems like it is still trying to call blockstream.info instead of the local Bitcoin API, I'm not sure if this is related but I can't seem to figure out why block stream is still being called, could use some help getting this first version over the finish line.

Everything is working as far as I can tell but I am having trouble getting BTC transferred to the right address with Clarinet, I've reached out to Ludo about that.

Want me to create a fork and then PR or grant me write access to the repo?
